File: twoSum.py
def twoSum(nums, target):
    for i in range(0, len(nums)):
        num1 = nums[i]
        new.remove(num1)
        for j in range(0, len(new)):
            if num1 + new[j] == target:
                sol.append(num1)
                sol.append(new[j])
            else:
                pass
    return sol


nums = [2, 7, 11, 15]
target = 18
new = []
sol = []

for i in range(0, len(nums)):
    new.append(nums[i])

print("nums : ", nums)
print("new : ", new)
twoSum(nums, target)
print(sol)

File: TwoSum.java
import java.util.Scanner;
import java.util.HashMap;
import java.util.Arrays;

class Solution 
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);
        
        System.out.print("Enter the array size : ");
        int size = sc.nextInt();
        int[] nums = new int[size];
        
        System.out.print("Enter the arry values : ");
        for (int i=0; i<size; i++)
        {
            nums[i]=sc.nextInt();
        }
        
        System.out.print("Enter target value : ");
        int target = sc.nextInt();
        
        System.out.print(Arrays.toString(twoSum(nums, target)));
        
    }
    public static int[] twoSum(int[] nums, int target) 
    {

        int[] arr = new int[2];
        HashMap <Integer, Integer> map = new HashMap<>();

        for (int i=0; i<nums.length; i++)
        {
            int complement = target - nums[i];

            if (map.containsKey(complement))
            {
                arr[0] = map.get(complement);
                arr[1] = i;
            }
            else
            {
                map.put(nums[i],i);
            }
        }
        return arr;
    }
}
